fideslog/api/models/analytics_event.py

```diff
@@ -81,12 +93,22 @@ def check_not_an_email_address(cls, value: str) -> str:
         return value
 
     @validator("endpoint")
-    def check_no_hostname(cls, value: str) -> str:
+    def validate_endpoint_format(cls, value: str) -> str:
         """
-        Ensure that endpoints contain only the URL path.
+        Ensure that endpoint contains method and path, and that path component contains only the URL path.
         """
 
-        return urlparse(value).path
+        endpoint_components = value.split(":")
+        assert (
+            len(endpoint_components) == 2
+        ), "endpoint must contain only the HTTP method and URL path, delimited by a colon"
+        http_method = endpoint_components[0].strip().upper()
+        path = endpoint_components[1].strip()
+        assert (
+            http_method in ALLOWED_HTTP_METHODS
+        ), f"HTTP method must be one of {', '.join(ALLOWED_HTTP_METHODS)}"
+        assert path == urlparse(path).path, "endpoint must contain only the URL path"
+        return f"{http_method}: {path}"
```

Comment:
Thinking more about this - do you think the user experience would be better or worse if we _didn't_ raise a `ValueError` for malformed URL paths? If instead this looked like:
```suggestion
        return f"{http_method}: {urlparse(path).path}"
```
We'd always capture just the URL path, but if a user included a full URL in their event, they wouldn't have any feedback as to why it was truncated. Thoughts?

Reply:
I'm more for explicit rather than implicit here. Raising the validation error initially might save some debugging time down the road, when someone sees the truncated form in the db.